docs/api.md

```diff
@@ -1799,6 +1804,20 @@ puppeteer.launch().then(async browser => {
 
 In the case of multiple pages in a single browser, each page can have its own viewport size.
 
+To avoid race condition with page's logic that sniffs for viewport properties,
```

Comment:
This sounds like inside baseball. Can we phrase it in a way that will make more sense to puppeteer users? Is this a bug or intended behavior? What happens if I emulate after navigating? If its a bug where is the link?

This also implies that I can't use setViewport to simulate a resize event. Is that true?